packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -167,6 +208,27 @@ export default createRule<Options, MessageId>({
         if (invalidAncestor.type === AST_NODE_TYPES.ReturnStatement) {
           // handle return statement
 
+          if (options.ignoreVoidInVoid) {
+            const functionNode = findFunction(invalidAncestor);
```

Comment:
[Bug] The following code won't be reported, because `findFunction` doesn't stop on function declarations:

```ts
type Foo = () => void;

const test1: Foo = function () {
  function testtt() {
    return console.log(); // not reported - bug
  }
};


const test2 = function () {
  function testtt() {
    return console.log(); // reported - nice
  }
};

```

[playground](https://deploy-preview-8632--typescript-eslint.netlify.app/play/#ts=5.3.3&fileType=.ts&code=C4TwDgpgBAYg9nKBeKAKAlMgfFAbnASwBMBuAKAGM4A7AZ2CmAnoC5YFkoAzAV2ouAEaaTAG8yUbnwFDqjZsEUYo4yZIBOEYD3VyqdOABsIAOkNwA5hnKSAvmVvkgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1rI6YDNYyZgHNaANw6UAJvQAexaCiG90YANrhsORNGgdokADSatWLdkiVhnRQDVJUgJJN701fjiIT2AL7eAupp%2BPkA&tsconfig=&tokens=false)